include/transform_output/transform_output.hpp

```diff
@@ -0,0 +1,215 @@
+			 * associated with time value read previously.
+			 */
+			if(line[0]=='[' & line[1]!=']'){
+
+				/**
+				 * We use strtok() to extract all tokens in the line by using delimiter ']'
+				 * The idea behind this is to separate out component and port from each other.
+				 * token_1 holds the first part of line and token_2 holds second part of line read from imput file.
+				 * e.g : token_1 = [iestream_input_defs<message_t>::out: {5}
+				 *       token_2 = generated by model generator_con
+				 */
+				char* token = strtok(line, "]"); 
+  				char* token_1 = token,* token_2 =strtok(NULL, "]");
+  				
+  				/**
+  				 * divide token_2 by delimiter space (" ") to extract component part from it.
+  				 * token_6 holds component part at the end of while loop.
+  				 */
+  				char* token_5 = strtok(token_2, " "), * token_6;
+  				
+  				/**
+			     * Keep printing tokens while one of the 
+			     * delimiters present in token_2. 
+			     */
+			    while (token_5 != NULL) { 
+			        component = token_5;
+			        token_5 = strtok(NULL, " ");
+			    } 
+			    
+			    /**
+			     * divide token_1 by delimiter comma (",") to extract all port and its corresponding values
+			     * while loop runs until all ports and its values are obtained
+			     * for each token retrieved in while loop, we check it with port constant to check if a port is
+			     * present in the token. If yes then we assign the port to string variable "port"
+			     */
+			    char* token_3 = strtok(token_1, ",");
+			    char* token_4;
+
+			    /**
+			     * Keep printing tokens while one of the 
+			     * delimiters present in token_3.
+			     */ 
+			    while (token_3 != NULL) { 
+
+			    	if(strstr(token_3,"out")){
+
+						port= "out";
+						
+					}else if(strstr(token_3,"dataOut")){
+
+						port="dataOut";
+
+					}else if(strstr(token_3,"packetSentOut")){
+
+					 	port="packetSentOut";
+
+					} else if(strstr(token_3,"ackReceivedOut")){
+
+					 	port="ackReceivedOut";
+					}
+					
+					/**
+					 * for each token_3, we extract the port value by using strstr() with "{" delimiter.
+					 * port value is present in a "{}" brackets.
+					 */
+					
+					token_4 = strstr(token_3,"{");
+					
+					/**
+				 	 * if ports value is empty then we ignore such ports and continue. This is verifed if the length of
+					 * token_4 is greater than 2.
+					 * if a value is obtained other than empty then we assign it to port_value variable.
+					 */
+		            if (strlen(token_4)>2){
+						
+						in = token_4;
+						port_value = in.substr( 1 , ( in.find_first_of('}', 0) - 1) );
+	        			
+	        			/**
+	        			 *Getting our data and writing it to output file.
+	        			 * if time is not zero then we write the line to output file.
+	        			 */ 
+	        			if (outputfile.is_open()){
+
+	        				if(time!="00:00:00:000"){

```

Comment:
When I ran the program, I noticed that for sender test the following line is omitted:

00:00:00:000
[iestream_input_defs<message_t>::out: {-1}] generated by model generator_con

Time Value Port Component
00:00:05:000 0 out generator_con

At the lecture the Professor clarified that 00:00:00:000 line is omitted only when
there are no value in it, but in this line there is a value of -1, so it should also
be included in the transform file